libobs/obs-output.c

```diff
@@ -2045,8 +2048,24 @@ static inline void convert_flags(const struct obs_output *output,
 	else
 		flags &= output->info.flags;
 
-	*has_video = (flags & OBS_OUTPUT_VIDEO) != 0;
-	*has_audio = (flags & OBS_OUTPUT_AUDIO) != 0;
+	if ((flags & OBS_OUTPUT_VIDEO) != 0) {
+		if ((flags & OBS_OUTPUT_OPTIONAL_VIDEO) != 0)
+			*has_video = output->video_encoder != NULL;
```

Comment:
This make the assumption that `OBS_OUTPUT_ENCODED` is set which is not a good thing.